### PR TITLE
Performance exercise set

### PR DESCRIPTION
**Pull request by samuelchassot**
- Correct `gradle` errors and update versions where needed
- various typo corrections
- add small tweaks here and there

### CODE REVIEW

File: exercises/performance/jmh-benchmarking/pom.xml [hunk 1]
```diff
@@ -60,12 +60,12 @@ THE POSSIBILITY OF SUCH DAMAGE.
         <!--
```
Comment (SolalPirelli) [inline exercises/performance/jmh-benchmarking/pom.xml:60]: Do we need a POM, which I assume means using Maven? The testing exercise repo had JMH using a Gradle plugin: https://github.com/sweng-epfl/testing-lecture-exercises/blob/main/build.gradle

File: exercises/performance/multi-threaded-server/build.gradle [hunk 2]
```diff
@@ -40,7 +41,7 @@ if (JavaVersion.current().toString() != javaVersionTarget) {
 }
 
 application {
-    mainClass = 'App'
```
Comment (SolalPirelli) [inline exercises/performance/multi-threaded-server/build.gradle:43]: Can you rename the SingleThreadedServer instead? We should try to keep build.gradle files as close to each other as possible. Same remark for the others.

Comment (samuelchassot) [reply exercises/performance/multi-threaded-server/build.gradle:43]: Yes I could but as they are supposed to change it while doing the exercise, we should find something more consistent. IMHO having `App` and then `AppMultiThreaded` does not make a lot of sense. Do you have other name ideas?

File: exercises/performance/jmh-benchmarking/pom.xml (hunk 1 above)
Comment (samuelchassot) [reply exercises/performance/jmh-benchmarking/pom.xml:60]: I'll change to use gradle

**Pull request closed without merging**